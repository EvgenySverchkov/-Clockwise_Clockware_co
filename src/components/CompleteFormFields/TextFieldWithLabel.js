import React from "react";

import FormGroup from "../FormComponents/FormGroup";
import Label from "../FormComponents/Label";
import TextField from "../FormComponents/TextField";

const TextFieldWithLabel = ({ fieldName, changeHandler, value }) => (
  <FormGroup key={fieldName} isRow={true}>
    <Label forId={fieldName}>Enter {fieldName}</Label>
    <TextField
      id={fieldName}
      name={fieldName}
      value={value || ""}
      chngHandler={changeHandler}
    />
  </FormGroup>
);

export default TextFieldWithLabel;
