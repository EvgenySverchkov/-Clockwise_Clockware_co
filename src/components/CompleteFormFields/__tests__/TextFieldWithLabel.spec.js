import React from "react";

import {create} from "react-test-renderer";
import TextFieldWithLabel from "../TextFieldWithLabel";
import ContextComponent from "../../../ContextComponent";

it("<TextFieldWithLabel /> test which is used in forms", ()=>{
    const mockProps = {
        fieldName: "FieldName", 
        changeHandler: jest.fn(), 
        value: "Value"
    }
    const mockContextValue = {
        warningTooltipMsg: "Message",
        isOpenWarningTooltip: false,
        idForTooltip: 1
    }
    const component = create(
        <ContextComponent.Provider value = {mockContextValue}>
            <TextFieldWithLabel {...mockProps}/>
        </ContextComponent.Provider>
    );
    expect(component.toJSON()).toMatchSnapshot();
})