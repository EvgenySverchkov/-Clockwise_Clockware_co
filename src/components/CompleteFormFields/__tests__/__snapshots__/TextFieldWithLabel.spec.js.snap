// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextFieldWithLabel /> test which is used in forms 1`] = `
<div
  className="form-group row text-center text-sm-left col-12 col-md-10 col-lg-8"
>
  <label
    className="pl-0 pr-0 col-sm-4 col-md-4 col-form-label font-weight-bold"
    htmlFor="FieldName"
  >
    Enter 
    FieldName
  </label>
  <div
    className="col-sm-8 col-md-8"
  >
    <input
      aria-describedby={null}
      className="form-control"
      id="FieldName"
      name="FieldName"
      onBlur={[Function]}
      onChange={[MockFunction]}
      onFocus={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      required={true}
      title="Message"
      type="text"
      value="Value"
    />
  </div>
</div>
`;
